assets/app/view/part/city.rb

```diff
@@ -145,13 +152,70 @@ def render_part
         children = []
         children << render_box(slots.size) if slots.size.between?(2, 6)
         children.concat(slots)
+
+        if (revenue = render_revenue)
+          children << revenue
+        end
+
         props = @city.solo? ? {} : { on: { click: -> { touch_node(@city) } } }
         h(:g, props, children)
       end
 
-      # TODOS:
-      # - do actual math and get points for the 3-slot hexagon, rather than
-      #   scaling the full-size hexagon
+      def render_revenue
+        revenues = @city.revenue.values.uniq
+        return if revenues.uniq.size > 1
+
+        revenue = revenues.first
+        return if revenue.zero?
+
+        # let View::Tile worry about rendering revenue if there are too many
+        # individual cities (eg, Chi in 1846)
+        return if @tile.cities.size > 2
+
+        angle = 0
+        displacement = 42
+        x = render_location[:x]
+        y = render_location[:y]
+        regions = []
+
+        # if there are more than 2 cities on the tile (e.g., Chi in 1846), the
+        # revenue should be handled by View::Tile
+        case @cities
+        when 1
+          x = 0
+          y = 0
+          regions = [11, 18]
+
+          case @city.slots
+          when 1
+            regions = [9, 16]
+          when 2
+            displacement = 67
+          when 3
+            displacement = 65
+          when 4
+            displacement = 67
+          end
+        when 2
+          if @edge
+            angle = OO_REVENUE_ANGLES[@edge]
+            regions = OO_REVENUE_REGIONS[@edge]
+          end
+        end
+
+        increment_weight_for_regions(regions)
+
+        h(:g, { attrs: { transform: "translate(#{x} #{y})" } }, [
```

Comment:
this is probably right, but why are there 3 nested svgs?

Reply:
Top nested svg is to get to the same spot on the tile as the city (since OO cities aren't in the middle). I first tried to do something in the main render function but it was pretty messy.

The next two svgs are my workaround/hack to do polar coordinates. If preferred, I could probably do the math to convert my (angle, distance) pair into (x, y) and knock out one of the `<g>` elements